vm/systemSmartContracts/stakingWaitingList.go

```diff
@@ -835,14 +837,55 @@ func (s *stakingSC) unStakeAllNodesFromQueue(args *vmcommon.ContractCallInput) v
 		// delete element from waiting list
 		inWaitingListKey := createWaitingListKey(blsKey)
 		s.eei.SetStorage(inWaitingListKey, nil)
+
+		ownerAddr := string(registrationData.OwnerAddress)
+		_, exists := mapOwnerKeys[ownerAddr]
+		if !exists {
+			mapOwnerKeys[ownerAddr] = make([][]byte, 0)
+			orderedListOwners = append(orderedListOwners, ownerAddr)
+		}
+
+		mapOwnerKeys[ownerAddr] = append(mapOwnerKeys[ownerAddr], blsKey)
 	}
 
 	// delete waiting list head element
 	s.eei.SetStorage([]byte(waitingListHeadKey), nil)
 
+	// call unStakeAtEndOfEpoch from the delegation contracts to compute the new unStaked list
```

Comment:
instead of the comment, we should split this into a new function. Related to @mariusmihaic comment.

Reply:
already made a split inside.